tokio/src/io/util/reader_stream.rs

```diff
@@ -0,0 +1,77 @@
+use crate::io::AsyncRead;
+use crate::stream::Stream;
+use bytes::{Bytes, BytesMut};
+use pin_project_lite::pin_project;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+pin_project! {
+    /// Convert async reader into stream of Result<Bytes, io::Error>
+    ///
+    /// This type can be created using the [`reader_stream`](crate::io::reader_stream) function
+    #[derive(Debug)]
+    #[cfg_attr(docsrs, doc(cfg(feature = "stream")))]
+    #[cfg_attr(docsrs, doc(cfg(feature = "io-util")))]
+    pub struct ReaderStream<R> {
+        // reader itself
+        #[pin]
+        reader: R,
+        // Working buffer, used to optimize allocations.
+        // # Capacity behavior
+        // Initially `buf` is empty. Also it's getting smaller and smaller
+        // during polls (because it's chunks are returned to stream user).
+        // But when it's capacity reaches 0, it is growed.
+        buf: BytesMut,
+    }
+}
+
+/// Convert `AsyncRead` into stream of byte chunks.
+/// # Example
+/// ```
```

Comment:
We need some newlines here.
```suggestion
/// Convert `AsyncRead` into stream of byte chunks.
///
/// # Example
///
/// ```
```